internal/controller/streams.go

```diff
@@ -328,6 +331,9 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 	}
 	logger.Info("Successfully sent resource snapshot")
 
+	// Signal that the snapshot commit has been sent
+	close(snapshotCommitSignal)
```

Comment:
Make sure that the channel is always closed, even if this function returns prematurely above, e.g. in case or error.
Otherwise, this code currently leaks channels.